From 75213e4374ff8db35c1faa9eb4813a0e6d46a826 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:01 -0400
Subject: [tools] perf/kvm: Introduce option -v for perf kvm command

Message-id: <1408953210-26343-114-git-send-email-jolsa@redhat.com>
Patchwork-id: 88819
O-Subject: [PATCH RHEL7.1 BZ1133083 113/322] perf kvm: Introduce option -v for perf kvm command.
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 100b907350c87aa1f3b5dbd95bac3ad5aad3e108
Author: Dongsheng Yang <yangds.fnst@cn.fujitsu.com>
Date: Mon Dec 9 12:15:11 2013 -0500

description
===========
As there is no -v option for perf kvm, the all debug message for perf
kvm will nerver be printed out to user.

Example:
 # perf kvm --guestmount /tmp/guestmount/ record -a
 Not enough memory for reading perf file header

It is confusing message for newbies such as me. With this patch applied,
we can use -v option to get the detail.

Example:
 # perf kvm --guestmount /tmp/guestmount/ record -a -v
 Can't access file /tmp/guestmount//15069/proc/kallsyms
 Not enough memory for reading perf file header
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Documentation/perf-kvm.txt b/tools/perf/Documentation/perf-kvm.txt
index 96a9a1d..52276a6 100644
--- a/tools/perf/Documentation/perf-kvm.txt
+++ b/tools/perf/Documentation/perf-kvm.txt
@@ -10,9 +10,9 @@ SYNOPSIS
 [verse]
 'perf kvm' [--host] [--guest] [--guestmount=<path>
  [--guestkallsyms=<path> --guestmodules=<path> | --guestvmlinux=<path>]]
- {top|record|report|diff|buildid-list}
+ {top|record|report|diff|buildid-list} [<options>]
 'perf kvm' [--host] [--guest] [--guestkallsyms=<path> --guestmodules=<path>
- | --guestvmlinux=<path>] {top|record|report|diff|buildid-list|stat}
+ | --guestvmlinux=<path>] {top|record|report|diff|buildid-list|stat} [<options>]
 'perf kvm stat [record|report|live] [<options>]
 
 DESCRIPTION
@@ -93,6 +93,9 @@ OPTIONS
  kernel module information. Users copy it out from guest os.
 --guestvmlinux=<path>::
  Guest os kernel vmlinux.
+-v::
+--verbose::
+ Be more verbose (show counter open errors, etc).
 
 STAT REPORT OPTIONS
 -------------------
diff --git a/tools/perf/builtin-kvm.c b/tools/perf/builtin-kvm.c
index f8bf5f2..d9cc0e3 100644
--- a/tools/perf/builtin-kvm.c
+++ b/tools/perf/builtin-kvm.c
@@ -1690,6 +1690,8 @@ int cmd_kvm(int argc, const char **argv, const char *prefix __maybe_unused)
       "file", "file saving guest os /proc/kallsyms"),
   OPT_STRING(0, "guestmodules", &symbol_conf.default_guest_modules,
       "file", "file saving guest os /proc/modules"),
+  OPT_INCR('v', "verbose", &verbose,
+       "be more verbose (show counter open errors, etc)"),
   OPT_END()
  };
 
-- 
1.7.1